airbyte-integrations/connectors/source-jira/source_jira/streams.py

```diff
@@ -288,17 +296,45 @@ def should_retry(self, response: requests.Response) -> bool:
         # for all other HTTP errors the default handling is applied
         return super().should_retry(response)
 
+    def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, Any]]]:
+        self._starting_point_cache.clear()
+        yield from read_full_refresh(self.boards_stream)
+
     def read_records(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
-        for board in read_full_refresh(self.boards_stream):
-            try:
-                yield from super().read_records(stream_slice={"board_id": board["id"]}, **kwargs)
-            except HTTPError as e:
-                if self._is_board_error(e.response):
-                    # Wrong board is skipped
-                    self.logger.warning(f"Board {board['id']} has no columns with a mapped status. Skipping.")
-                    continue
-                else:
-                    raise
+        try:
+            yield from super().read_records(stream_slice={"board_id": stream_slice["id"]}, **kwargs)
+        except HTTPError as e:
+            if self._is_board_error(e.response):
+                # Wrong board is skipped
+                self.logger.warning(f"Board {stream_slice['id']} has no columns with a mapped status. Skipping.")
+            else:
+                raise
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]):
+        updated_state = latest_record[self.cursor_field]
+        board_id = str(latest_record["boardId"])
+        stream_state_value = current_stream_state.get(board_id, {}).get(self.cursor_field)
+        if stream_state_value:
+            updated_state = max(updated_state, stream_state_value)
+        current_stream_state.setdefault(board_id, {})[self.cursor_field] = updated_state
+        return current_stream_state
+
+    def get_starting_point(self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any]) -> Optional[pendulum.DateTime]:
+        board_id = str(stream_slice["board_id"])
+        if self.cursor_field not in self._starting_point_cache:
+            self._starting_point_cache.setdefault(board_id, {})[self.cursor_field] = self._get_starting_point(
+                stream_state=stream_state, stream_slice=stream_slice
+            )
+        return self._starting_point_cache[board_id][self.cursor_field]
+
+    def _get_starting_point(self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any]) -> Optional[pendulum.DateTime]:
+        board_id = str(stream_slice["board_id"])
+        if stream_state:
```

Comment:
you don't need assign board_id before if statement. It's used only `stream_state` not false.